accepted/targeting-packs-and-runtime-packs.md

```diff
@@ -0,0 +1,287 @@
+
+**Proposal**
+ * Targeting pack packages: \<shared framework name\>.Ref
+ * Runtime pack packages: \<shared framework name\>.Runtime.\<rid\>
+ * AppHost pack packages: \<shared framework name\>.Host.\<rid\>
+
+**Examples**
+ * Microsoft.NETCore.App.Ref ("base" targeting pack for base .NET Core)
+ * Microsoft.ASPNetCore.App.Ref (targeting pack for ASP.NET Core)
+ * Microsoft.WindowsDesktop.App.Ref (targeting pack for WPF/WinForms on .NET Core)
+ * NETStandard.Library.Ref (targeting pack for .NET Standard)
+ * Microsoft.NETCore.App.Runtime.linux-x64 ("base" runtime pack for .NET Core for Linux OS and x64 CPU)
+ * Microsoft.NETCore.App.Host.linux-x64 (apphost pack for .NET Core for Linux OS and x64 CPU)
+ * Microsoft.ASPNetCore.App.Runtime.osx-x64 (runtime pack for ASP.NET Core for Mac OS and x64 CPOU)
+ * Microsoft.WindowsDesktop.App.Runtime.win-x86 (targeting pack for WPF/Winforms on .NET Core for Windows OS and x64 CPU)
+
+
+## Package Versions
+
+Runtime pack package versions will be 1:1 with .NET Core Runtime / shared framework versions. This follows from the fact that runtime packs contain shared framework implementation and thus must change whenever the implementation changes. 
+
+Targeting pack package versions will generally not increase past major.minor.0 where major.minor matches the the corresponding two-part TFM. For example, when there is a 3.0.1 .NET Core runtime, the targeting pack will likely remain at 3.0.0. This follows from the fact that targeting packs represent public API surface, which must not change in a patch version of the runtime. (With that said, we can reserve the right to modify the targeting pack in a patch release to fix a severe bug. In the rare event that this occurs, the targeting pack patch version could be incremented past 0.)
+
+During the prerelease phase of a new major.minor version of the runtime, the targeting pack will version 1:1 with the runtime. This is necessary as different prerelease versions will have different surface area.
+
+For example, 
+
+* .NET Core SDK 3.0.100-preview with 3.0.0-preview runtime
+    * packs/Microsoft.NETCore.App.Ref/3.0.0-preview
+    * shared/Microsoft.NETCore.App/3.0.0-preview
+
+* .NET Core SDK 3.0.100 with 3.0.0 runtime
+    * packs/Microsoft.NETCore.App.Ref/3.0.0
+    * shared/Microsoft.NETCore.App/3.0.0
+
+* .NET Core SDK 3.0.101 with 3.0.1 runtime
+    * packs/Microsoft.NETCore.App.Ref/3.0.0
+    * shared/Microsoft.NETCore.App/3.0.1
+
+
+## Package Layout
+
+Since the packages will have a special type and not be installable into projects in the usual way, we do not need to use folders in the nupkg like `ref/\<TFM>` etc. Instead we will simply place the pack contents in the root of the .nupkg. The package names and versions already uniquely identify the content per TFM/RID and a big part of the package size savings comes from avoiding assets that span multiple TFMs in a single package.
```

Comment:
We can't put the files in the root because older clients will interpret this as assets to use. Any package type enforcement is limited to newer clients. Also, it is probably too confusing to have everything in the root.

Current proposal is a new generic package type "Data" and folder data/. This means that the package is used purely for transport and the contents of data/ are opaque to Nuget and up to the transporter to consume appropriately. This could be reused by other environments such as CoreXT with the same requirements.

Reply:
But see also https://github.com/dotnet/designs/pull/50#discussion_r254865137

Reply:
We're going with single new package type that isn't overly generic: DotnetPlatform. Mirrors DotnetTool package layout with standard nuget conventions: e.g. ref/netcoreapp3.0/(reference assemblies), runtimes/(rid)/lib/(tfm)/(impl assemblies), runtimes/(rid)/native/(native binaries).

We will use data/ for things that are given to the SDK with no equivalent in nuget: platform manifest, package overrides, file list.